Sources/NIO/SelectorEpoll.swift

```diff
@@ -171,12 +171,11 @@ extension Selector: _SelectorBackendProtocol {
             ready = Int(try Epoll.epoll_wait(epfd: self.selectorFD, events: events, maxevents: Int32(eventsCapacity), timeout: -1))
         }
 
-        // start with no deregistrations happened
-        self.deregistrationsHappened = false
-        // temporary workaround to stop us delivering outdated events; possibly set in `deregister`
-        for i in 0..<ready where !self.deregistrationsHappened {
+        for i in 0..<ready {
             let ev = events[i]
-            switch ev.data.fd {
+            let fd = Int32(UInt64(ev.data.u64 & 0x00000000FFFFFFFF))
```

Comment:
We don't need the UInt64 constructor and we can safely just use `truncatingIfNeeded` here. Similarly, we should use `CInt` instead of `Int32`.

```suggestion
            let fd = CInt(truncatingIfNeeded: ev.data.u64)
```